mailbox/api/src/main/java/org/apache/james/mailbox/model/UpdatedFlags.java

```diff
@@ -36,6 +38,45 @@
  */
 public class UpdatedFlags {
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private MessageUid uid;
+        private Flags oldFlags;
+        private Flags newFlags;
+        private Optional<Long> modSeq = Optional.absent();
+
```

Comment:
I concider it is boiler plate and do not add much value IMO.

Making the constructor of the built class private is a hudge win as you force using the builder, with explicit naming, and force builder adoption. Here, the choice between builder constructor and static method is just about fluency (they are strictly equivalent).

Anyway, while squashing, I will make it private.